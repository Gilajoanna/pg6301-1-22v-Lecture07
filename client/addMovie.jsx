import * as React from "react";
import { useState } from "react";
import { postJSON } from "./postJSON";

function FormInput({ label, value, onChangeValue }) {
  return (
    <div>
      <label>{label}</label>{" "}
      <input
        type={"text"}
        name={"title"}
        value={value}
        onChange={(e) => onChangeValue(e.target.value)}
      />
    </div>
  );
}

export function AddNewMovie({ createMovie }) {
  const [title, setTitle] = useState("");
  const [year, setYear] = useState();
  const [country, setCountry] = useState("");
  const [plot, setPlot] = useState("");

  async function handleSubmit(e) {
    e.preventDefault();
    createMovie({ title });
    await postJSON("/api/movies/new", {
      title,
      year,
      countries: [country],
      plot,
    });
    setTitle("");
    setYear("");
    setPlot("");
    setCountry("");
  }

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <h1>Add new movie</h1>

        <FormInput label={"Title:"} value={title} onChangeValue={setTitle} />
        <FormInput label={"Year:"} value={year} onChangeValue={setYear} />
        <FormInput
          label={"Country:"}
          value={country}
          onChangeValue={setCountry}
        />
        <FormInput label={"Plot:"} value={plot} onChangeValue={setPlot} />
        <button type={"submit"}>Submit</button>
      </form>
    </div>
  );
}
