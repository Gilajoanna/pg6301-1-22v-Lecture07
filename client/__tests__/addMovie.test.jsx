import * as React from "react";
import * as ReactDOM from "react-dom";

import { AddNewMovie } from "../addMovie";
import { Simulate } from "react-dom/test-utils";

describe("add movie component", () => {
  it("shows movies form", () => {
    const element = document.createElement("div");
    ReactDOM.render(<AddNewMovie />, element);
    expect(element.innerHTML).toMatchSnapshot();
    expect(
      Array.from(element.querySelectorAll("form label")).map((e) => e.innerHTML)
    ).toEqual(["Title:", "Year:", "Country:", "Plot:"]);
  });

  it("adds movie on submit", () => {
    const createMovie = jest.fn();
    const title = "Test Movie";
    const element = document.createElement("div");
    ReactDOM.render(<AddNewMovie createMovie={createMovie} />, element);
    Simulate.change(element.querySelector("form input:nth-of-type(1)"), {
      target: { value: title },
    });
    Simulate.submit(element.querySelector("form"));
    expect(createMovie).toBeCalledWith({
      title,
    });
  });
});
