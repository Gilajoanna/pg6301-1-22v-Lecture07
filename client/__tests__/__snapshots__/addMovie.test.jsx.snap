// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add movie component shows movies form 1`] = `"<div><form><h1>Add new movie</h1><div><label>Title:</label> <input type=\\"text\\" name=\\"title\\" value=\\"\\"></div><div><label>Year:</label> <input type=\\"text\\" name=\\"title\\" value=\\"\\"></div><div><label>Country:</label> <input type=\\"text\\" name=\\"title\\" value=\\"\\"></div><div><label>Plot:</label> <input type=\\"text\\" name=\\"title\\" value=\\"\\"></div><button type=\\"submit\\">Submit</button></form></div>"`;
